src/stores/AccountController.ts

```diff
@@ -12,78 +16,129 @@ import type {
+  private readonly checkoutService: CheckoutService;
+  private readonly accountService: AccountService;
+  private readonly subscriptionService: SubscriptionService;
+  private readonly favoritesController: FavoritesController;
+  private readonly watchHistoryController: WatchHistoryController;
+  private readonly profileController?: ProfileController;
+
+  constructor(
+    checkoutService: CheckoutService,
+    accountService: AccountService,
+    subscriptionService: SubscriptionService,
+    favoritesController: FavoritesController,
+    watchHistoryController: WatchHistoryController,
+    profileController?: ProfileController,
+  ) {
+    this.checkoutService = checkoutService;
+    this.accountService = accountService;
+    this.subscriptionService = subscriptionService;
+    this.favoritesController = favoritesController;
+    this.watchHistoryController = watchHistoryController;
+    this.profileController = profileController;
+  }
+
+  loadUserData = async () => {
     try {
-      const authData = await accountService.getAuthData();
+      const authData = await this.accountService.getAuthData();
 
       if (authData) {
-        await getAccount();
-        await restoreWatchHistory();
-        await restoreFavorites();
+        await this.getAccount();
+        await this.watchHistoryController?.restoreWatchHistory();
+        await this.favoritesController?.restoreFavorites();
       }
     } catch (error: unknown) {
       logDev('Failed to get user', error);
 
       // clear the session when the token was invalid
       // don't clear the session when the error is unknown (network hiccup or something similar)
       if (error instanceof Error && error.message.includes('Invalid JWT token')) {
-        await logout();
+        await this.logout();
       }
     }
-  });
-};
-
-export const initializeAccount = async () => {
-  await useService(async ({ accountService, config }) => {
-    if (!accountService) {
-      useAccountStore.setState({ loading: false });
-      return;
-    }
+  };
 
+  initializeAccount = async () => {
     useAccountStore.setState({
       loading: true,
-      canUpdateEmail: accountService.canUpdateEmail,
-      canRenewSubscription: accountService.canRenewSubscription,
-      canManageProfiles: accountService.canManageProfiles,
-      canUpdatePaymentMethod: accountService.canUpdatePaymentMethod,
-      canChangePasswordWithOldPassword: accountService.canChangePasswordWithOldPassword,
-      canExportAccountData: accountService.canExportAccountData,
-      canDeleteAccount: accountService.canExportAccountData,
-      canShowReceipts: accountService.canShowReceipts,
     });
+    const config = useConfigStore.getState().config;
 
-    loadPersistedProfile();
+    const features = this.accountService.features;
 
-    await accountService.initialize(config, logout);
+    useAccountStore.setState({
+      features: {
+        hasIntegration: true,
+        canDeleteAccount: features.canExportAccountData,
```

Comment:
I don't think we need to duplicate the feature settings into the store. If they are a readonly property on the service, we can use them anywhere needed in the controllers. For any components that need them we can either add a getter to the appropriate controller or just let the component drill down from the controller to the service.

Reply:
Not all of them are readonly props. `canManageProfiles` is a dynamic one.

Reply:
Looking at that one, it looks like a buggy approach. The service determines if profiles are potentially enabled, then we should handle other logic based on whether there are profiles. I would suggest we don't rewrite the feature setting.